### PR TITLE
Extra comment data

### PR DESCRIPTION
**Pull request by proyectoramirez**
Adds extra comment data to the list of comments from DataServlet. This is then used while populating the comments on the client.

Demo: https://extra-comment-data-dot-josedramirez-step-2020.appspot.com/

This can be merged in no particular order.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        String content = (String) entity.getProperty("content");
+        long timestamp = (long) entity.getProperty("timestamp");
+        double sentimentScore = (double) entity.getProperty("sentimentScore");
+        double sentimentMagnitude = (double) entity.getProperty("sentimentMagnitude");
+
+        Comment comment = new Comment(content, timestamp, sentimentScore, sentimentMagnitude);
+
+        return comment;
+    }
+}
```
Comment (Enriqe) [inline portfolio/src/main/java/com/google/sps/data/Comment.java]:
```suggestion
}

```

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -336,11 +336,55 @@ function buildCommentFragment(comment) {
...
+ * 
+ * @param {number} sentimentScore Number in range [-1, 1], which will determine
+ *    the hue of the color.
+ * @param {number} sentimentMagnitude Number in range [0, 1], which will determine
+ *    the lightness of the color
+ * 
+ * @return {string} A string containing a CSS color
+ */
+function getSentimentColor(sentimentScore, sentimentMagnitude) {
+  const hue = mapRange(sentimentScore, -1, 1, 0, 120);
```
Comment (Enriqe) [inline portfolio/src/main/webapp/script.js]: can you add comments to these arguments that are a bit ambiguous?

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -336,11 +336,55 @@ function buildCommentFragment(comment) {
...
+  return `hsl(${hue}, 100%, ${lightness}%)`;
+}
+
+/**
+ * Maps a number in a given range to a value in different range.
+ * All ranges are inclusive.
+ * 
+ * @param {number} value The number to map.
+ * @param {number} in1 The lower bound of the input range.
+ * @param {number} out1 The upper bound of the input range.
```
Comment (Enriqe) [inline portfolio/src/main/webapp/script.js]:
```suggestion
 * @param {number} lowerBoundIn The lower bound of the input range.
 * @param {number} upperBoundIn The upper bound of the input range.
```

**Review by Enriqe — COMMENTED** [review]
Woahh 💯

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (proyectoramirez) [reply portfolio/src/main/webapp/script.js]: Done in latest commit!

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (proyectoramirez) [reply portfolio/src/main/webapp/script.js]: Done!

**Review by Enriqe — APPROVED** [review]
Awesome!! 👏

**Pull request merged**